llarp/rpc/rpc_server.cpp

```diff
@@ -107,626 +61,518 @@ namespace llarp::rpc
+    {
+      SetJSONError("Invalid remote: " + lookupsnode.request.routerID, lookupsnode.response);
+      return;
+    }
+
+    m_Router.loop()->call([&]() {
+      auto endpoint = m_Router.exitContext().GetExitEndpoint("default");
+
+      if (endpoint == nullptr)
+      {
+        SetJSONError("Cannot find local endpoint: default", lookupsnode.response);
+        return;
+      }
+
+      endpoint->ObtainSNodeSession(routerID, [&](auto session) {
+        if (session and session->IsReady())
+        {
+          const auto ip = net::TruncateV6(endpoint->GetIPForIdent(PubKey{routerID}));
+          util::StatusObject status{{"ip", ip.ToString()}};
+          SetJSONResponse(status, lookupsnode.response);
+          return;
+        }
+
+        SetJSONError("Failed to obtain snode session", lookupsnode.response);
+        return;
+      });
+    });
+  }
+
+  void
+  RPCServer::invoke(MapExit& mapexit)
+  {
+    MapExit exit_request;
+    // steal replier from exit RPC endpoint
+    exit_request.replier.emplace(mapexit.move());
+
+    m_Router.hiddenServiceContext().GetDefault()->map_exit(
+        mapexit.request.address,
+        mapexit.request.token,
+        mapexit.request.ip_range,
+        [exit = std::move(exit_request)](bool success, std::string result) mutable {
+          if (success)
+            exit.send_response({{"result"}, std::move(result)});
+          else
+            exit.send_response({{"error"}, std::move(result)});
+        });
+  }
+
+  void
+  RPCServer::invoke(ListExits& listexits)
+  {
+    if (not m_Router.hiddenServiceContext().hasEndpoints())
+    {
+      SetJSONError("No mapped endpoints found", listexits.response);
+      return;
+    }
+
+    auto status = m_Router.hiddenServiceContext().GetDefault()->ExtractStatus()["exitMap"];
+
+    SetJSONResponse((status.empty()) ? "No exits" : status, listexits.response);
+  }
+
+  void
+  RPCServer::invoke(UnmapExit& unmapexit)
+  {
+    try
+    {
+      m_Router.routePoker()->Down();
+      for (auto& ip : unmapexit.request.ip_range)
+        m_Router.hiddenServiceContext().GetDefault()->UnmapExitRange(ip);
+    }
+    catch (std::exception& e)
+    {
+      SetJSONError("Unable to unmap to given range", unmapexit.response);
+      return;
+    }
+
+    SetJSONResponse("OK", unmapexit.response);
+  }
+
+  //  Sequentially calls map_exit and unmap_exit to hotswap mapped connection from old exit
+  //  to new exit. Similar to how map_exit steals the oxenmq deferredsend object, swapexit
+  //  moves the replier object to the unmap_exit struct, as that is called second. Rather than
+  //  the nested lambda within map_exit making the reply call, it instead calls the unmap_exit logic
+  //  and leaves the message handling to the unmap_exit struct
+  void
+  RPCServer::invoke(SwapExits& swapexits)
+  {
+    MapExit map_request;
+    UnmapExit unmap_request;
+    auto current_exits = m_Router.hiddenServiceContext().GetDefault()->ExtractStatus()["exitMap"];
```

Comment:
```suggestion
    auto endpoint = m_Router.hiddenServiceContext().GetDefault();
    auto current_exits = endpoint->ExtractStatus()["exitMap"];
```

and then also replace `m_Router.hiddenServiceContext().GetDefault()` the 2 places below with `endpoint` (and capture `endpoint` in the lambda by value)